x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -0,0 +1,262 @@
+			defer a.workerWg.Done()
+			defer a.workerCancel()
+			var err error
+			if a.config.EPHEnabled {
+				err = a.runWithEPH()
+			} else {
+				err = a.run()
+			}
+			if err != nil {
+				a.log.Error(err)
+				return
+			}
+		}()
+	})
+}
+
+// run runs the input
+func (a *azureInput) run() error {
+	persister, err := persist.NewFilePersister(fmt.Sprintf("%s/azureventhubpersister", paths.Data))
+	if err != nil {
+		return err
+	}
+	hubs, err := eventhub.NewHubFromConnectionString(fmt.Sprintf("%s%s%s", a.config.ConnectionString, eventHubConnector, a.config.EventHubName), eventhub.HubWithOffsetPersistence(persister))
+	//ctx, cancel := context.WithCancel(a.workerCtx)
+	//defer cancel()
+	if err != nil {
+		return err
+	}
+	a.hub = hubs
+
+	// listen to each partition of the Event Hub
+	runtimeInfo, err := a.hub.GetRuntimeInformation(a.workerCtx)
+	if err != nil {
+		return err
+	}
+
+	var receiveOption eventhub.ReceiveOption
+	if a.config.ConsumerGroup != "" {
+		receiveOption = eventhub.ReceiveWithConsumerGroup(a.config.ConsumerGroup)
+	}
+	for _, partitionID := range runtimeInfo.PartitionIDs {
+		// Start receiving messages
+		//
+		// Receive blocks while attempting to connect to hub, then runs until listenerHandle.Close() is called
+		// <- listenerHandle.Done() signals listener has stopped
+		// listenerHandle.Err() provides the last error the receiver encountered
+		_, err := a.hub.Receive(
+			a.workerCtx,
+			partitionID,
+			func(c context.Context, event *eventhub.Event) error {
+				return a.processEvents(event, partitionID)
+			},
+			receiveOption)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Wait for a signal to quit:
+
+	//err = hub.Close(context.Background())
+	//if err != nil {
+	//	fmt.Println(err)
+	//}
+
+	return nil
+}
+
+// Stop stops TCP server
+func (a *azureInput) Stop() {
+	if a.hub != nil {
+		err := a.hub.Close(a.workerCtx)
+		if err != nil {
+			a.log.Errorw(fmt.Sprintf("error while closing eventhub"), "error", err)
+		}
+	}
+	if a.processor != nil {
+		err := a.processor.Close(a.workerCtx)
+		if err != nil {
+			a.log.Errorw(fmt.Sprintf("error while closing eventhostprocessor"), "error", err)
+		}
+	}
+	a.workerCancel()
+	a.workerWg.Wait()
+}
+
+// Wait stop the current server
+func (a *azureInput) Wait() {
+	a.Stop()
+}
+
+func (a *azureInput) processEvents(event *eventhub.Event, partitionID string) error {
+	// timestamp temp disabled as the event date is applied for now, will be replaced
+	//timestamp := time.Now()
+	var events []beat.Event
+	azure := common.MapStr{
+		"partition_id":   partitionID,
+		"eventhub":       a.config.EventHubName,
+		"consumer_group": a.config.ConsumerGroup,
+	}
+	messages := a.parseMultipleMessages(event.Data)
+	for _, msg := range messages {
+		for key, value := range msg {
+			azure.Put("offset", event.SystemProperties.Offset)
+			azure.Put("sequence_number", event.SystemProperties.SequenceNumber)
+			azure.Put("enqueued_time", event.SystemProperties.EnqueuedTime)
```

Comment:
where is `event.SystemProperties` set?

Reply:
the event has type eventhub.Event, those properties are set in the azure-eventhub dependencies

Reply:
thanks for clarifying! these fields need to be added to `fields.yml` so they are correctly mapped, see https://github.com/elastic/beats/blob/master/filebeat/_meta/fields.common.yml#L123 for example